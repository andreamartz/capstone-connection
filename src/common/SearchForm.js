import React, { useState } from "react";
import Alert from '@material-ui/lab/Alert';
import { useHistory } from 'react-router-dom';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  form: {
    width: '100%', // Fix IE 11 issue.
    display: 'flex'
  },
  textField: {
    flexGrow: '1'
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

/** Search form 
 * 
 * Appears on PrjList so that projects shown can be filtered.
 * 
*/

const SearchForm = ({ search }) => {
  console.debug("SearchForm", "searchFor=", typeof search);

  const [searchTerm, setSearchTerm] = useState("");
  const [formErrors, setFormErrors] = useState([]);

  console.debug("SearchForm",
    "search=", typeof search,
    "searchTerm=", searchTerm,
    "formErrors", formErrors
  );

  const classes = useStyles();
  const history = useHistory();

  /** Tell parent component to do the filtering */
  const handleSubmit = (evt) => {
    // prevent accidentally searching just spaces
    evt.preventDefault();
    search(searchTerm.trim() || undefined);
    setSearchTerm(searchTerm.trim());
  };

  /** Update form field as user types */
  const handleChange = (evt) => {
    setSearchTerm(evt.target.value);
  }

  return (
    <Paper className={classes.paper} elevation={5}>
      <form className={classes.form} 
        onSubmit={handleSubmit}>
        <TextField className={classes.textField}
          name="searchTerm"
          onChange={handleChange}
          value={searchTerm}
          variant="outlined"
        />
        <Button type="submit"
          variant="contained"
          color="primary"
        >
          Search
        </Button>
      </form>
    </Paper>
  );
}

export default SearchForm;