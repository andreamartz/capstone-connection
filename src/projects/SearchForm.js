import React, { useState } from "react";
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Paper from '@material-ui/core/Paper';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  paper: {
    width: '100%',
    marginRight: theme.spacing(2)
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    display: 'flex'
  },
  textField: {
    flexGrow: '1'
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
}));

/** Search form 
 * 
 * Appears above PrjCardList so that projects shown can be filtered.
 * 
*/

const SearchForm = ({ search }) => {
  console.debug("SearchForm", "searchFor=", typeof search);

  const [searchTerm, setSearchTerm] = useState("");
  const [formErrors, setFormErrors] = useState([]);

  console.debug("SearchForm",
    "searchTerm=", searchTerm,
    "formErrors", formErrors
  );

  const classes = useStyles();

  /** Tell parent component to do the filtering */
  const handleSubmit = (evt) => {
    // prevent accidentally searching just spaces
    evt.preventDefault();
    search(searchTerm.trim() || undefined);
    setSearchTerm("");
  };

  /** Update form field as user types */
  const handleChange = (evt) => {
    setSearchTerm(evt.target.value);
  }

  return (
    <Paper className={classes.paper} elevation={5}>
      <form className={classes.form} 
        onSubmit={handleSubmit}>
        <TextField className={classes.textField}
          name="searchTerm"
          onChange={handleChange}
          value={searchTerm}
          variant="outlined"
        />
        <Button type="submit"
          variant="contained"
          color="primary"
        >
          Search by Tag
        </Button>
      </form>
    </Paper>
  );
}

export default SearchForm;