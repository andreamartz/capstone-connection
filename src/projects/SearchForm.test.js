import React from 'react';
import { render } from '@testing-library/react';
import SearchForm from './SearchForm';

test('it renders without crashing', function () {
	render(<SearchForm searchTerm={'JS'} />);
});

test('it matches snapshot', function () {
	const { asFragment } = render(<SearchForm searchTerm={'JS'} />);
	expect(asFragment()).toMatchSnapshot();
});
