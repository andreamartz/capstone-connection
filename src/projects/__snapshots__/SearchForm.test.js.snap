// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-14 makeStyles-box-10"
  >
    <form
      class="makeStyles-form-11"
    >
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-textField-12"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            name="searchTerm"
            placeholder="Search by tag"
            type="text"
            value="JS"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-15 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-16"
              style="width: 0.01px;"
            >
              <span>
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
